packages/core-components/src/layout/Sidebar/Items.tsx

```diff
@@ -82,128 +82,133 @@ export type SidebarItemClassKey =
   | 'arrows'
   | 'selected';
 
-const useStyles = makeStyles<BackstageTheme>(
-  theme => {
-    const {
+const useStyles = () => {
+  const {
+    sidebarConfig: {
       selectedIndicatorWidth,
       drawerWidthClosed,
       drawerWidthOpen,
       iconContainerWidth,
-    } = sidebarConfig;
-    return {
-      root: {
-        color: theme.palette.navigation.color,
-        display: 'flex',
-        flexFlow: 'row nowrap',
-        alignItems: 'center',
-        height: 48,
-        cursor: 'pointer',
-      },
-      buttonItem: {
-        background: 'none',
-        border: 'none',
-        width: '100%',
-        margin: 0,
-        padding: 0,
-        textAlign: 'inherit',
-        font: 'inherit',
-      },
-      closed: {
-        width: drawerWidthClosed,
-        justifyContent: 'center',
-      },
-      open: {
-        [theme.breakpoints.up('sm')]: {
-          width: drawerWidthOpen,
+    },
+  } = useContext(SidebarConfigContext);
+
+  return makeStyles<BackstageTheme>(
```

Comment:
I'm worried this is gonna have a big impact on performance. Could you perhaps grab the values from context and then pass them in to `useStyles` instead?

Reply:
:+1: